### ISSUE TITLE
[load-themed-styles] Style duplication when using composes from less.

### ISSUE THREAD

**Issue by Markionium** [microsoft/web-build-tools#787]
TLDR; `loader-load-themed-styles`, depending on setup, adds duplicate styles to the DOM. Would you be interested in adopting a similar approach to `style-loader` and ignore/replace duplicate styles?

Hi all,

We've recently started using `load-themed-styles` and `loader-load-themed-styles`.

The `loader-load-themed-styles` states the following
> It is designed to be a replacement for style-loader.

I'd like to make you aware that this statement makes it seem like replacing `style-loader` would be safe to do. However depending on your setup that is not necessarily true. (Which unfortunately resulted in a pretty significant escalation on our side.)

One significant difference is that `style-loader` keeps track of which styles it added to the DOM and does not add duplicates of identical styles.

https://github.com/webpack-contrib/style-loader/blob/1ca12ab32221933aece24e0be9122e1a3bcf5d40/lib/addStyles.js#L123-L125 

`load-themed-styles` does not keep track of which styles are added to the DOM. This can result in duplicate style rules. (In the image below shows just one duplicate, but in our production product we've seen the same style up to 8 times.)

![image](https://user-images.githubusercontent.com/2134901/44708559-bbb10100-aaa7-11e8-9863-6f4f90c61b02.png)

In our situation this happens when using CSS modules and composing styles using less. I have created a small repro of it. https://github.com/Markionium/repro-dual-style-load

The reason this happens is that `loadStyles` is called with the composed class twice and it adds both to the DOM. Below is a screenshot of the calls to `loadStyles` in my repro.
![image](https://user-images.githubusercontent.com/2134901/44708764-35e18580-aaa8-11e8-868c-3b5016814a45.png)

As a solution we've wrapped `loader-load-themed-styles` and pass our wrapper loader that also removes the duplicate styles.

```ts
const LoadThemedStylesLoader = require('@microsoft/loader-load-themed-styles');

LoadThemedStylesLoader.loadedThemedStylesPath = "./LoadThemedStylesLoaderThatIgnoresDuplicates";
```

Would this duplicate ignoring functionality be something you'd be interested in merging into `loader-load-themed-styles` or `load-themed-styles`? If so i'd be willing to take look at creating a PR for it.

**Comment by pgonzal** [comment]
@dzearing @iclanton

**Comment by pgonzal** [comment]
@dzearing @iclanton could we find someone familiar with **load-themed-styles** to answer this question?
